src/app/BasicRule.cpp

```diff
@@ -0,0 +1,202 @@
+                    pSpecs.emplace_back(paraIndex, action, spec["value"].as<std::vector<int>>());
+                }
+                else {
+                    // don't need extra check yet
+                    pSpecs.emplace_back(paraIndex, action, spec["value"].as<int>());
+                }
+
+                // build specs, potentially
+                if (!needExtraCheck) {
+                    const scmp_datum_t value = spec["value"].as<scmp_datum_t>();
+                    specs.push_back((struct scmp_arg_cmp){paraIndex, this->cmpActionMap.at(action), value});
+                }
+            }
+        }
+        if (!needExtraCheck) {
+            this->rules.emplace_back(sysnum, id, specs);
+        }
+        else {
+            this->rules.emplace_back(sysnum, id, pSpecs);
+        }
+    }
+
+    spdlog::info("init BasicRule module");
+    this->initRules();
+}
+
+void BasicRule::initRules() {
+    int offset = 0;
+    for (Rule rule : this->rules) {
+        if (rule.needExtraCheck) {
+            seccomp_rule_add(*this->ctxp, SCMP_ACT_TRACE(SM_EVM_BASIC_BASE + offset), rule.sysnum, 0);
+        }
+        else {
+            switch (rule.specs.size()) {
+                case 0:
+                    seccomp_rule_add(*this->ctxp, SCMP_ACT_TRACE(SM_EVM_BASIC_BASE + offset), rule.sysnum, 0);
+                    break;
+                case 1:
+                    seccomp_rule_add(*this->ctxp, SCMP_ACT_TRACE(SM_EVM_BASIC_BASE + offset), rule.sysnum, 1, 
+                        rule.specs[0]);
+                    break;
+                case 2:
+                    seccomp_rule_add(*this->ctxp, SCMP_ACT_TRACE(SM_EVM_BASIC_BASE + offset), rule.sysnum, 2, 
+                        rule.specs[0], rule.specs[1]);
+                    break;
+                case 3:
+                    seccomp_rule_add(*this->ctxp, SCMP_ACT_TRACE(SM_EVM_BASIC_BASE + offset), rule.sysnum, 3, 
+                        rule.specs[0], rule.specs[1], rule.specs[2]);
+                    break;
+                case 4:
+                    seccomp_rule_add(*this->ctxp, SCMP_ACT_TRACE(SM_EVM_BASIC_BASE + offset), rule.sysnum, 4, 
+                        rule.specs[0], rule.specs[1], rule.specs[2], rule.specs[3]);
+                    break;
+                case 5:
+                    seccomp_rule_add(*this->ctxp, SCMP_ACT_TRACE(SM_EVM_BASIC_BASE + offset), rule.sysnum, 5, 
+                        rule.specs[0], rule.specs[1], rule.specs[2], rule.specs[3], rule.specs[4]);
+                    break;
+                case 6:
+                    seccomp_rule_add(*this->ctxp, SCMP_ACT_TRACE(SM_EVM_BASIC_BASE + offset), rule.sysnum, 6, 
+                        rule.specs[0], rule.specs[1], rule.specs[2], rule.specs[3], rule.specs[4], rule.specs[5]);
+                    break;
+            }
+        }
+        offset++;
+    }
+}
+
+bool BasicRule::check(const long eventMsg, const user_regs_struct &regs, const int tid) {
```

Comment:
this function is too lang. please split it into several small slices.

Reply:
fixed